sympy/matrices/normalforms.py

```diff
@@ -0,0 +1,115 @@
+from sympy.matrices import *
+
+'''
+Functions returning normal forms of matrices
+
+'''
+def smith_normal_form(m, domain = None):
```

Comment:
Can you add a docstring for this function along with information regarding type of inputs and returned values?
Few examples to go along with this would be ideal.